packages/lesswrong/components/comments/CommentWithReplies.tsx

```diff
@@ -16,68 +16,74 @@ const styles = (theme: ThemeType): JssStyles => ({
+  initialMaxChildren?: number;
+  commentNodeProps?: Partial<CommentsNodeProps>;
+  classes: ClassesType;
+}
 
-  if (!comment || !post)
-    return null;
+const CommentWithReplies = ({
+  comment,
+  post,
+  lastRead,
+  markAsRead = () => {},
+  initialMaxChildren = 3,
+  commentNodeProps,
+  classes,
+}: CommentWithRepliesProps) => {
+  const [maxChildren, setMaxChildren] = useState(initialMaxChildren);
+  
+  if (!comment) return null;
+  
+  const lastCommentId = comment.latestChildren[0]?._id;
+  
+  const treeOptions: CommentTreeOptions = {
+    lastCommentId,
+    markAsRead: markAsRead,
+    highlightDate: lastRead,
+    condensed: true,
+    showPostTitle: true,
+    post,
+    ...(commentNodeProps?.treeOptions || {}),
+  };
 
-  const lastCommentId = comment.latestChildren[0]?._id
+  const { CommentsNode } = Components;
 
-  const renderedChildren = comment.latestChildren.slice(0, maxChildren)
-  const extraChildrenCount = (comment.latestChildren.length > renderedChildren.length) && (comment.latestChildren.length - renderedChildren.length)
+  const renderedChildren = comment.latestChildren.slice(0, maxChildren);
+  const extraChildrenCount =
+    comment.latestChildren.length > renderedChildren.length && comment.latestChildren.length - renderedChildren.length;
 
-  let nestedComments = unflattenComments(renderedChildren)
+  let nestedComments = unflattenComments(renderedChildren);
   if (extraChildrenCount > 0) {
-    nestedComments = addGapIndicators(nestedComments)
+    nestedComments = addGapIndicators(nestedComments);
   }
 
-  const lastVisitedAt = markedAsVisitedAt || post.lastVisitedAt
-
-  const showExtraChildrenButton = (extraChildrenCount>0) ? 
-    <a className={classes.showChildren} onClick={()=>setMaxChildren(500)}>
-      Showing 3 of {comment.latestChildren.length } replies (Click to show all)
-    </a> : null
+  const showExtraChildrenButton =
+    extraChildrenCount > 0 ? (
+      <a className={classes.showChildren} onClick={() => setMaxChildren(maxChildren + 500)}>
+        Showing {maxChildren} of {comment.latestChildren.length} replies (Click to show all)
+      </a>
+    ) : null;
 
-  const treeOptions: CommentTreeOptions = {
-    lastCommentId,
-    markAsRead: markAsRead,
-    highlightDate: lastVisitedAt,
-    condensed: true,
-    showPostTitle: showTitle,
-    refetch,
-    post,
-  };
-  
-  return <CommentsNode
-    treeOptions={treeOptions}
-    noHash
-    startThreadTruncated={true}
-    nestingLevel={1}
-    comment={comment}
-    childComments={nestedComments}
-    key={comment._id}
-    shortform
-    expandByDefault={expandByDefault}
-    showExtraChildrenButton={showExtraChildrenButton}
-  />
-}
+  return (
+    <CommentsNode
+      noHash
+      startThreadTruncated={true}
+      nestingLevel={1}
+      comment={comment}
+      childComments={nestedComments}
+      key={comment._id}
+      shortform
+      showExtraChildrenButton={showExtraChildrenButton}
+      {...commentNodeProps}
```

Comment:
This was just passing a lot of props through to CommentsNode so I refactored it to allow passing arbitrary props through